src/chromo/common.py

```diff
@@ -732,9 +774,7 @@ def cross_section(self, kin=None):
                 for component, fraction in zip(kin2.p2.components, kin2.p2.fractions):
                     kin3.p2 = component
                     # this calls cross_section recursively, which is fine
-                    cs = self.cross_section(kin3)
-                    for i, val in enumerate(dataclasses.astuple(cs)):
-                        cross_section[i] += fraction * val
+                    cross_section += fraction * self.cross_section(kin3)
```

Comment:
I suggest to revert the change to CrossSectionData. You are replacing three lines of readable code here are only required in one place with over 30 lines. That is only an improvement if we expect to do multiply and add on CrossSectionData in many places, which I don't see happening.

Reply:
I don’t like how it was written before. It is only technically readable but boilerplate wrt to the physics… dataclasses.astuple is not readable, IMO.

My only concern about the operator implementations is that they have to list the fields by hand, apart from the names for the cross sections, which are aligned with Pythia’s internal naming scheme.

Reply:
Actually one more concern here, why the operators don’t follow the implementation of what has been done in the other methods?? Iteration over .astuple?

Reply:
dataclasses.astuple is the official way to handle a tuple-like dataclass. It is better and more readable to use this than setattr and getattr like in the replacement code.

Reply:
> It is only technically readable but boilerplate wrt to the physics…

The change replaces this mathematical three-liner which expresses compactly what is happening mathematically here with a generic API that locks CrossSectionData into the role of a value-like object that supports addition and multiplication. Effectively, CrossSectionData now acts like a numpy array.

That would be ok, if that's what we need in many places, and if users should be able to multiplity and add CrossSectionData objects. I don't see that. And if there is no general use case for this functionality, it should **not** define the public API of the class. (I missed a "not" in my original comment in a crucial place...).

Public API you cannot easily take back later. From a maintenance point of view, we have to be stingy about what is exposed publically, only expose what is really needed for sure.

Here, we are not doing this. We expose a public API for functionality that we only use internally and in only one place. It is a bad trade-off.

Reply:
> dataclasses.astuple is the official way to handle a tuple-like dataclass. It is better and more readable to use this than setattr and getattr like in the replacement code.

I used setattr and getattr because  `'CrossSectionData' object is not subscriptable`. I decided to add `__getitem___` and `__setitem__`  to make it  subscriptable and increase readability. In this case we don't need + and += operators.

The other concern to address when + and += operators were introduced was that in the future CrossSectionData might have fields which should not be added. When such fields will appear `dataclasses.fields(cs)` should be changed to the function which returns only appropriate fields.

Reply:
> The other concern to address when + and += operators were introduced was that in the future CrossSectionData might have fields which should not be added. When such fields will appear dataclasses.fields(cs) should be changed to the function which returns only appropriate fields.

I also think that we may add further fields that should not be added, and that is a reason why I am against adding operators to CrossSectionData at this point.

From the point of future maintainability, I can see a case for moving the code for adding a scaled copy into the class itself. That way, if someone changes the class in the future, they are more likely to notice that this code has to be changed as well.

A solution which does not expose this to users is to add a private method to CrossSectionData like so:
```py
class CrossSectionData:
     ...
     def _mul_add(self, factor, other):
           for field in self.fields:
                 setattr(self, field, factor * getattr(other, field))
```

Reply:
Sorry, I did not respond to the issue with `'CrossSectionData' object is not subscriptable`. I am a bit surprised by this, because I thought the original code passed CI tests. I was under the wrong impression that a dataclass is subscriptable, but I was wrong. So we indeed need setattr and getattr, that is correct.

Reply:
_mul_radd is used now.